File: Projeto-MobiFix/Database/dtos/serviceDtos/intervencaoCatalogoDto.js
const paraIntervencaoCatalogoDto = (i) => {
    if (!i) return null;
    return {
        IntervencaoID: i._id,
        Descricao: i.descricao,
        PrecoFixoMaoDeObra: i.precoFixoMaoDeObra,
        Especialidade: i.especialidade
    };
};

module.exports = { paraIntervencaoCatalogoDto };

File: Projeto-MobiFix/Backend/Models/Catalogos/IntervencaoCatalogoDto.cs
namespace Backend.Models;

using System.ComponentModel.DataAnnotations;

public class IntervencaoCatalogoDto
{
    public int IntervencaoID { get; set; }
    public string Descricao { get; set; } = string.Empty;
    public decimal PrecoFixoMaoDeObra { get; set; }
    public string? Especialidade { get; set; }
}

public class IntervencaoCatalogoCriacaoDto
{
    [Required(ErrorMessage = "A descrição é obrigatória.")]
    public string Descricao { get; set; } = string.Empty;

    [Range(0, double.MaxValue, ErrorMessage = "O preço deve ser positivo.")]
    public decimal PrecoFixoMaoDeObra { get; set; }

    public string? Especialidade { get; set; }
}